homeassistant/components/remote_rpi_gpio/switch.py

```diff
@@ -0,0 +1,93 @@
+"""Allows to configure a switch using RPi GPIO."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.components import remote_rpi_gpio
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['remote_rpi_gpio']
+
+CONF_ADDRESS = 'address'
+CONF_PULL_MODE = 'pull_mode'
+CONF_PORTS = 'ports'
+CONF_INVERT_LOGIC = 'invert_logic'
+
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ADDRESS): cv.string,
+    vol.Required(CONF_PORTS): _SWITCHES_SCHEMA,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Remote Raspberry PI GPIO devices."""
+    address = config.get(CONF_ADDRESS)
+    invert_logic = config.get(CONF_INVERT_LOGIC)
+
+    switches = []
+    ports = config.get(CONF_PORTS)
+    for port, name in ports.items():
+        switches.append(RemoteRPiGPIOSwitch(name, address, port, invert_logic))
+    add_entities(switches)
+
+
+class RemoteRPiGPIOSwitch(ToggleEntity):
```

Comment:
Inherit from `SwitchDevice` imported from the switch base component instead.

Reply:
Forgive my ignorance, but I'm not sure how to do what you're asking.  I was trying to mimic the raspberrypi gpio component since this is very similar, just on a remote system.

Reply:
```py
from homeassistant.components.switch import SwitchDevice

...

class RemoteRPiGPIOSwitch(SwitchDevice):
```